blockchain/message_pool/src/config.rs

```diff
@@ -0,0 +1,78 @@
+use address::Address;
+use db::Store;
+use encoding::{from_slice, to_vec};
+use serde::{Deserialize, Serialize};
+use std::error::Error as StdError;
+use std::time::Duration;
+
+const MPOOL_CONFIG_KEY: &[u8] = b"/mpool/config";
+const SIZE_LIMIT_LOW: i64 = 20000;
+const SIZE_LIMIT_HIGH: i64 = 30000;
+const PRUNE_COOLDOWN: Duration = Duration::from_secs(60); // 1 minute
+const REPLACE_BY_FEE_RATIO: f64 = 1.25;
+const GAS_LIMIT_OVERESTIMATION: f64 = 1.25;
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct MpoolConfig {
+    priority_addrs: Vec<Address>,
+    size_limit_high: i64,
+    size_limit_low: i64,
+    replace_by_fee_ratio: f64,
+    prune_cooldown: Duration,
+    gas_limit_overestimation: f64,
+}
+
+impl Default for MpoolConfig {
+    fn default() -> Self {
+        Self {
+            priority_addrs: vec![],
+            size_limit_high: SIZE_LIMIT_HIGH,
+            size_limit_low: SIZE_LIMIT_LOW,
+            replace_by_fee_ratio: REPLACE_BY_FEE_RATIO,
+            prune_cooldown: PRUNE_COOLDOWN,
+            gas_limit_overestimation: GAS_LIMIT_OVERESTIMATION,
+        }
+    }
+}
+
+impl MpoolConfig {
+    pub fn new(
+        priority_addrs: Vec<Address>,
+        size_limit_high: i64,
+        size_limit_low: i64,
+        replace_by_fee_ratio: f64,
+        prune_cooldown: Duration,
+        gas_limit_overestimation: f64,
+    ) -> Result<Self, String> {
+        // Validate if parameters are valid
+        if replace_by_fee_ratio < REPLACE_BY_FEE_RATIO {
+            return Err(format!(
+                "replace_by_fee_ratio:{} is less than required: {}",
+                replace_by_fee_ratio, REPLACE_BY_FEE_RATIO
+            ));
+        }
+        if gas_limit_overestimation < 1.0 {
```

Comment:
why does this not use the `GAS_LIMIT_OVERESTIMATION` const?

Reply:
because its validating the params to make a new config